### ISSUE TITLE
Support bidirectional type inference for args with dictionary unpacking

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5358]
In cases where an argument to a call expression uses dictionary unpacking and the target is a function that accepts a `**kwargs` typed as an unpacked `TypedDict`, it would be nice to use bidirectional type inference to infer the type of a literal dictionary expression.

```python
from typing import TypedDict, Unpack

class FuncOptions(TypedDict, total=False):
    opt1: bool

def func(code: str | None = None, **options: Unpack[FuncOptions]):
    pass

func(**{"opt1": True}) # Currently generates an error
```

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.316, which I just published. It will also be included in this week's insiders build of pylance.
